packages/openid4vci/src/credential-request/z-credential-response.ts

```diff
@@ -3,28 +3,49 @@ import { zOauth2ErrorResponse } from '../../../oauth2/src/common/z-oauth2-error'
 
 const zCredentialEncoding = z.union([z.string(), z.record(z.string(), z.any())])
 
-export const zCredentialResponse = z
+const zBaseCredentialResponse = z
   .object({
     credential: z.optional(zCredentialEncoding),
     credentials: z.optional(z.array(zCredentialEncoding)),
 
+    interval: z.number().int().positive().optional(),
+
+    notification_id: z.string().optional(),
+  })
+  .passthrough()
+
+export const zCredentialResponse = zBaseCredentialResponse
+  .extend({
     transaction_id: z.string().optional(),
 
     c_nonce: z.string().optional(),
     c_nonce_expires_in: z.number().int().optional(),
-
-    notification_id: z.string().optional(),
   })
   .passthrough()
-  .refine(
-    (value) => {
-      const { credential, credentials, transaction_id } = value
-      return [credential, credentials, transaction_id].filter((i) => i !== undefined).length === 1
-    },
-    {
-      message: `Exactly one of 'credential', 'credentials', or 'transaction_id' MUST be defined.`,
+  .superRefine((value, ctx) => {
+    const { credential, credentials, transaction_id, interval, notification_id } = value
+
+    if ([credential, credentials, transaction_id].filter((i) => i !== undefined).length !== 1) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: `Exactly one of 'credential', 'credentials', or 'transaction_id' MUST be defined.`,
+      })
     }
-  )
+
+    if (transaction_id && !interval) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: `'interval' MUST be defined when 'transaction_id' is defined.`,
+      })
+    }
```

Comment:
ℹ️ Actually only required from Draft 16. If we want to support non-draft 16 deferred credential issuance, we need to remove this rule.

Reply:
I think it's fine to only support Draft 16 deferred issuance

Reply:
Okay, then I can even streamline the response and remove the singular `credential` as option.